src/main/java/com/shopping_app/shoppingApp/controller/ProductControler.java

```diff
@@ -0,0 +1,53 @@
+package com.shopping_app.shoppingApp.controller;
+
+import com.shopping_app.shoppingApp.model.Enum.ResponseType;
+import com.shopping_app.shoppingApp.model.Request.ProductRequest;
+import com.shopping_app.shoppingApp.model.Response.ApiResponse;
+import com.shopping_app.shoppingApp.model.Response.ProductResponse;
+import com.shopping_app.shoppingApp.service.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/product/")
+@Validated
+@RequiredArgsConstructor(onConstructor = @__({@Autowired}))
+public class ProductControler {
```

Comment:
I will not be reviewing the product apis related code in this MR, so it would be helpful if you can comment it for now and work on it later when we move to this apis